### PR TITLE
Use unpublish endpoint

### PR DESCRIPTION
**Pull request by steventux**
https://trello.com/c/78Q9mrDe/708-make-whitehall-use-unpublishing-endpoint-medium

Make Whitehall use Publishing API `/unpublish` endpoint.

Unpublishing content in Whitehall should make an unpublishing request to the Publishing API.
This covers redirected, gone and withdrawn content.

### CODE REVIEW

File: test/unit/data_hygiene/organisation_reslugger_test.rb [hunk 1]
```diff
@@ -30,47 +30,33 @@ def teardown
...
         stub_publishing_api_publish(content_item.content_id, locale: 'en', update_type: 'major')
       ]
 
-      expected_redirect_requests = [
-        stub_publishing_api_put_content(redirect_item.content_id, redirect_item.content),
-        stub_publishing_api_patch_links(redirect_item.content_id, links: redirect_item.links),
-        stub_publishing_api_publish(redirect_item.content_id, locale: 'en', update_type: 'major')
-      ]
+      request = stub_publishing_api_unpublish(
+        @organisation.content_id,
```
Comment (gpeng) [inline test/unit/data_hygiene/organisation_reslugger_test.rb]: Is this redirecting the content item we've just published?

Comment (steventux) [inline test/unit/data_hygiene/organisation_reslugger_test.rb]: Good point, if the slug is updated before the redirect is issued the content item published by the slug update will be redirected, this isn't right. The redirect unpublishing needs to happen before the slug update. This runs the risk of redirecting then failing to update (the slug) leaving a broken redirect on the router though...

File: app/workers/publishing_api_gone_worker.rb [hunk 2]
```diff
@@ -1,7 +1,9 @@
...
-  def call(base_path, options = {})
-    draft = options.fetch("draft", false)
-    gone_item = PublishingApiPresenters::Gone.new(base_path)
-    draft ? save_draft(gone_item) : send_item(gone_item, 'en')
+  def call(content_id, locale)
+    Whitehall.publishing_api_v2_client.unpublish(
+      content_id,
+      type: "gone",
+      locale: locale,
+    )
```
Comment (danielroseman) [inline app/workers/publishing_api_gone_worker.rb]: I think this was the only thing that uses PublishingApiPresenters::Gone, so we should probably remove it.

**Comment by boffbowsh** [comment]
This LGTM 👍

**Pull request merged**